### DISCUSSION TITLE
Need help with my custom JetBot

### DISCUSSION THREAD

**Discussion by newcgen** [NVIDIA-AI-IOT/jetbot#409] [Q&A]
Hi!
I have read the Bill of Materials and assembled my custom JetBot.
Via SSH python3 works fine:
>>> from jetbot import Robot
>>> robot = Robot()         
>>> robot.left(speed=0.3)
>>> robot.stop()

But via JupyterLab it does not works. OSError: [Errno 121] Remote I/O error
The error is only when working with i2c.

Thanx!

**Comment by tjp1940** [comment]
Did you prepare and load the jetbot image card per the software
instructions?
If so can you assure the display is connected and working? The jetbot
program checks for i2c inputs
from the display and motor controller, and if not found the program just
loops and waits.  I had a bad display
 and that is what happened to me.  When I replaced the display all was ok.
TP

**Reply by newcgen** [reply]
The display is working (it shows ip,ram,disk).
 i2cdetect -y -r 1 shows display and PCA9685.

**Reply by tjp1940** [reply]
I have not used the JupyterLab that much. I suppose their i2c error could refer to some other.  When I had my bad display I connected an HMDI monitor up and the terminal screen showed the error message repeating every few seconds.  Just a thought......are you using a PCA9685 as a motor controller?  Also does your error message refer to hex address 0x3C ?

**Reply by newcgen** [reply]
jetbot@jetbot:~$ i2cdetect -y -r 1 
3c
40
70
The impression that JupyterLab does not have enough rights.

**Reply by tjp1940** [reply]
From that I would say your display (address 3c) and motor controller are up and running and the problem is within the jetbot and jupyter.  Sorry I do not have anything else to offer.

**Reply by Hoehlenbaer** [reply]
Do you have sample code for your robot? I compared the code of the dependencies of my third party kit with the GITHub version of the Adafruit motor controller and found the kit one heavily modified regarding addresses and channels. I'll try to add these modifications to the code of the jetpack files tonight. Let's see what happens...

**Comment by newcgen** [comment]
Only сhanging the physical address of PCA9685 to 0x60 helped.
